From 5ec5e294226ef8ef47af1b56b4c0c349aea7d607 Mon Sep 17 00:00:00 2001
From: Rui Matos <tiagomatos@gmail.com>
Date: Wed, 21 Sep 2016 15:06:08 +0200
Subject: [PATCH] clutter/stage: Fix framebuffer capture origin offset (again)

Commit 5fbb4793017b1891f0de4720a9793e8cfc1c8483 was wrong too. What we
really want to do here is getting view relative coordinates given the
view's and the rectangle's global coordinates so we need to subtract
the view's origin from the rectangle's.

https://bugzilla.gnome.org/show_bug.cgi?id=771502
---
 clutter/clutter/clutter-stage.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/clutter/clutter/clutter-stage.c b/clutter/clutter/clutter-stage.c
index 1d2c0bc..21968b9 100644
--- a/clutter/clutter/clutter-stage.c
+++ b/clutter/clutter/clutter-stage.c
@@ -4712,8 +4712,8 @@ capture_view (ClutterStage          *stage,
   clutter_stage_view_get_layout (view, &view_layout);
 
   cogl_framebuffer_read_pixels_into_bitmap (framebuffer,
-                                            view_layout.x + rect->x,
-                                            view_layout.y + rect->y,
+                                            rect->x - view_layout.x,
+                                            rect->y - view_layout.y,
                                             COGL_READ_PIXELS_COLOR_BUFFER,
                                             bitmap);
 
-- 
2.7.4

